src/lib/operators/table_scan/column_vs_column_table_scan_impl.cpp

```diff
@@ -28,66 +29,99 @@ std::string ColumnVsColumnTableScanImpl::description() const { return "ColumnVsC
 
 std::shared_ptr<PosList> ColumnVsColumnTableScanImpl::scan_chunk(ChunkID chunk_id) const {
   const auto chunk = _in_table->get_chunk(chunk_id);
+  const auto& left_segment = *chunk->get_segment(_left_column_id);
+  const auto& right_segment = *chunk->get_segment(_right_column_id);
+
+  std::shared_ptr<PosList> result;
+
+  // If the left and the right segment and/or type are not the same, we erase the types even for the release build.
```

Comment:
Also, which types? Iterator type? DataType? ComparatorType?

Reply:
"If the left and the right segment and/or data type are not the same" ?

Reply:
+ "we erase the **segment** types ..."

Reply:
the **segment iterable** types, no?

Reply:
see, the devil's in the details.